contracts/Overtime/FreeBetsHolder.sol

```diff
@@ -0,0 +1,164 @@
+    using SafeERC20 for IERC20;
+
+    uint private constant MAX_APPROVAL = type(uint256).max;
+
+    ISportsAMMV2 public sportsAMM;
+
+    ILiveTradingProcessor public liveTradingProcessor;
+
+    mapping(address => mapping(address => uint)) public balancePerUserAndCollateral;
+
+    mapping(address => bool) public supportedCollateral;
+
+    mapping(address => address) public ticketToUser;
+
+    mapping(address => uint) public paidPerTicket;
+
+    mapping(bytes32 => address) public liveRequestsPerUser;
+
+    /* ========== CONSTRUCTOR ========== */
+
+    function initialize(address _owner, address _sportsAMMV2, address _liveTradingProcessor) external initializer {
+        setOwner(_owner);
+        sportsAMM = ISportsAMMV2(_sportsAMMV2);
+        liveTradingProcessor = ILiveTradingProcessor(_liveTradingProcessor);
+    }
+
+    function fund(address _user, address _collateral, uint _amount) external notPaused {
+        require(supportedCollateral[_collateral], "Unsupported collateral");
+        IERC20(_collateral).safeTransferFrom(msg.sender, address(this), _amount);
+        balancePerUserAndCollateral[_user][_collateral] += _amount;
+        emit UserFunded(_user, _collateral, _amount, msg.sender);
+    }
+
+    function trade(
+        ISportsAMMV2.TradeData[] calldata _tradeData,
+        uint _buyInAmount,
+        uint _expectedPayout,
+        uint _additionalSlippage,
+        address _referrer,
+        address _collateral
+    ) external notPaused canTrade(msg.sender, _collateral, _buyInAmount) {
+        balancePerUserAndCollateral[msg.sender][_collateral] -= _buyInAmount;
+        address createdTicket = sportsAMM.trade(
+            _tradeData,
+            _buyInAmount,
+            _expectedPayout,
+            _additionalSlippage,
+            address(0),
+            _referrer,
+            _collateral == address(sportsAMM.defaultCollateral()) ? address(0) : _collateral,
+            false
+        );
+        ticketToUser[createdTicket] = msg.sender;
+    }
+
+    function tradeLive(
+        bytes32 _gameId,
+        uint16 _sportId,
+        uint16 _typeId,
+        uint8 _position,
+        uint _buyInAmount,
+        uint _expectedPayout,
+        uint _additionalSlippage,
+        address _referrer,
+        address _collateral
+    ) external notPaused canTrade(msg.sender, _collateral, _buyInAmount) {
+        bytes32 _requestId = liveTradingProcessor.requestLiveTrade(
+            _gameId,
+            _sportId,
+            _typeId,
+            _position,
+            _buyInAmount,
+            _expectedPayout,
+            _additionalSlippage,
+            address(0),
+            _referrer,
+            _collateral
+        );
+        liveRequestsPerUser[_requestId] = msg.sender;
+    }
+
+    function confirmLiveTrade(
+        bytes32 requestId,
+        address _createdTicket,
+        uint _buyInAmount,
+        address _collateral
+    ) external notPaused {
```

Comment:
seems like everyone can confirmLiveTrade for any user. 
Maybe that is the goal, just to check that

Reply:
great catch, it should be callable just by LiveTradingProcessor